.github/workflows/add-netlify-links.yml

```
@@ -0,0 +1,58 @@
+name: Add Netlify Links To Changed Pages
+on:
+  workflow_call:
+  pull_request_target:
+jobs:
+  get-pr-changes:
+    name: Get Changed Files & Update PR Description
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      contents: write
+      pull-requests: write
+      repository-projects: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get Changed Files
+        id: changed-files
+        # pin to a specific commit to ensure stability
+        uses: tj-actions/changed-files@c65cd883420fd2eb864698a825fc4162dd94482c
+        with:
+          separator: ","
+          files: source/**
+      - name: Build Netlify Links for Changed Pages
+        id: build_page_links
+        env:
+          CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
+        run: |
+          new_links=""
+          base_link='https://deploy-preview-${{ github.event.number }}--mongodb-docs-cpp.netlify.app'
```

updated to match C++ staging link

```suggestion
          base_link='https://deploy-preview-${{ github.event.number }}--docs-cpp.netlify.app'
```